plugins/http_plugin/include/eosio/http_plugin/macros.hpp

```diff
@@ -7,29 +7,81 @@ struct async_result_visitor : public fc::visitor<fc::variant> {
-                     std::get<fc::exception_ptr>(result)->dynamic_rethrow_exception();                           \
-                  } catch (...) {                                                                                \
-                     http_plugin::handle_exception(#api_name, #call_name, body, cb);                             \
-                  }                                                                                              \
-               } else if (std::holds_alternative<call_result>(result)) {                                      \
-                  cb(http_response_code, fc::time_point::maximum(), fc::variant(std::get<call_result>(result))); \
-               } else {                                                                                          \
-                  assert(0);                                                                                     \
-               }                                                                                                 \
-            });                                                                                                  \
-      } catch (...) {                                                                                            \
-         http_plugin::handle_exception(#api_name, #call_name, body, cb);                                         \
-      }                                                                                                          \
-   }                                                                                                             \
+#define CALL_ASYNC_WITH_400(api_name, api_handle, api_namespace, call_name, call_result, http_resp_code, params_type) \
+{ std::string("/v1/" #api_name "/" #call_name),                                                                 \
+  [api_handle, &_http_plugin](string&&, string&& body, url_response_callback&& cb) mutable {                    \
+     auto deadline = api_handle.start();                                                                        \
+     try {                                                                                                      \
+        auto params = parse_params<api_namespace::call_name ## _params, params_type>(body);                     \
+        FC_CHECK_DEADLINE(deadline);                                                                            \
+        using http_fwd_t = std::function<chain::t_or_exception<call_result>()>;                                 \
+        api_handle.call_name( std::move(params),                                                                \
+           [&_http_plugin, cb=std::move(cb), body=std::move(body)]                                              \
+           (const chain::next_function_variant<call_result>& result) {                                          \
```

Comment:
This should be `mutable` as `body` is moved below.

Reply:
good catch, thanks!